frontend/modules/tree/treeProcessing/transforming.ts

```diff
@@ -13,7 +13,7 @@ const getNamespacedId = (node) => {
 
 const getTransformedNodeData = (node) => ({
 	_id: node._id,
-	name: node.name,
+	name: node.name || node._id,
```

Comment:
@dleszcz @simplypixi Any insights? :)

Reply:
@carmenfan Ok, I'll do this

Reply:
@carmenfan
We were pretty sure it should work that way. xD

1. Why is this not handled on the BE side?
2. Should we hide list item only or get rid it of from the data?
3. What if the tree node (I mean node on the list) is a model and all of the child nodes don't have the name?

Reply:
1. Because it's not really appropriate for the BE can handle. A transformation may have multiple mesh children with no name, there's also situations where there's a transformation with a nameless mesh and other transformation children, so it's impossible for us to merge the 2 together.

2. Hide it from view, so when the user clicks on a nameless mesh, it should highlight it's parent (and any siblings)

3. only nameless meshes should be hidden. nameless transformations should be shown with `(No Name)`

Reply:
@carmenfan Done

Also rebased staging and I see that Santiago made changes in groups colors overriding. Unfortunately, he used old angularjs Tree service :/ 

I fixed this old TreeService usage and it looks like it works well with our migrated tree module, but I noticed also some watcher in _viewer.component.ts_ to update colors overrides.  We shouldn't use Angular anymore. Can you @sanmont3drepo migrate this?
By the way - why the color overriding is moved from saga to helpers? Now is impossible to use selectors in a proper way, and I had to use migration _getState()_ function.

Reply:
> @carmenfan Done
> 
> Also rebased staging and I see that Santiago made changes in groups colors overriding. Unfortunately, he used old angularjs Tree service :/
> 
> I fixed this old TreeService usage and it looks like it works well with our migrated tree module, but I noticed also some watcher in _viewer.component.ts_ to update colors overrides. We shouldn't use Angular anymore. Can you @sanmont3drepo migrate this?
> By the way - why the color overriding is moved from saga to helpers? Now is impossible to use selectors in a proper way, and I had to use migration _getState()_ function.

Hi Daniel, the way I've change the groups was to make the separation between state and the UI, reacting to the changes in the state as React does. This actually allows to use selectors as the transformation of the state ( as you can see as the selectors added for groups overrides) having  one source of truth instead of changing the override list in every interaction and trying to sync searches, overrides and edition of groups which tends to have problems (that's why we've chosen redux, right?). 

**The angular part only uses the suscription to see the difference between two states and call to the appropiate unity functions in an imperative way**.

I truly tried to patch it up using calls directly in the sagas instead of the other approach but it didn't work. Both worlds kind of collide.

 This goes in hand which the thing I've told @simplypixi  long ago about having something that actually represents the viewer as a UI layer instead of making calls in the sagas to unity functions.

I can change and try to push that part of angular into the react group's component, which is not the best because it doesn't represent the actual unity viewer but I think I'd work.
 
The only potential problem that I see is the getNodesFromSharedIds if you put that in the sagas part instead of a service.

Let me think if you think that there is another way to migrate this. I have no problem in doing it.

Reply:
Ok, so let's leave it now, it works fine. We'll finish migrating it after merging these changes with #1590